calnex/api/api.go

```diff
@@ -273,28 +272,22 @@ func parseResponse(response string) (string, error) {
 }
 
 // NewAPI returns an pointer of API struct with default values.
-func NewAPI(protocol APIProto, source string) *API {
-	transport := &http.Transport{}
-	if protocol == HTTPS {
-		transport = &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: false},
-		}
-	}
-
+func NewAPI(source string, strictTLS bool) *API {
```

Comment:
It's a nit, but I feel the API should have strict TLS by default, and thus it should be something like this.
It's a minor thing, but with boolean variables defaulting to `false` current version just reads wrong to me.
```
func NewAPI(source string, insecureTLS bool) *API {
```